src/transformers/trainer.py

```diff
@@ -200,6 +200,8 @@
             save_fsdp_model,
             save_fsdp_optimizer,
         )
+    if version.parse(accelerate_version) > version.parse("0.23.0"):
+        from accelerate.data_loader import SeedableRandomSampler
```

Comment:
```suggestion
    DATA_SAMPLERS = [RandomSampler]
    if version.parse(accelerate_version) > version.parse("0.23.0"):
        from accelerate.data_loader import SeedableRandomSampler
        DATA_SAMPLERS += [SeedableRandomSampler]
```
let's maybe only check for accelerate version once and define this here ?

Reply:
Makes sense!